torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -60,6 +62,30 @@ std::string getNcclVersion() {
   return versionString;
 }
 
+#ifdef USE_C10D_NCCL
+size_t hashTensors(const std::vector<at::Tensor>& tensors) {
+  size_t hash = 0;
+  for (auto& tensor : tensors) {
+    if (tensor.numel() > 0 && tensor.storage()) {
+      size_t data_size = tensor.storage().nbytes();
+      if (data_size > 0 && tensor.storage().data_ptr()) {
+        std::hash<char> hasher;
+        auto src = static_cast<const char*>(tensor.storage().data_ptr().get());
+        char* dst = (char*)std::calloc(data_size, sizeof(char));
+        cudaMemcpy(dst, src, data_size, cudaMemcpyDeviceToHost);
+        for (size_t i = 0; i < data_size; ++i) {
+          // Update the hash for each byte in the tensor
+          hash ^=
+              hasher(((char*)dst)[i]) + 0x9e3779b9 + (hash << 6) + (hash >> 2);
```

Comment:
https://stackoverflow.com/questions/74805446/how-to-hash-a-pytorch-tensor

also i think @ezyang made up some kind of tensor summary that wasn't a real hash but was fast.  In this case i think its worth doing a real hash but let me see if i can find that just in case.